lib/i18n.rb

```diff
@@ -231,6 +231,34 @@ def translate!(key, **options)
     end
     alias :t! :translate!
 
+    # Returns an array of interpolation keys for the given translation key
+    #
+    # Suppose we have the following:
+    #   I18n.t 'example.zero' == 'Zero interpolations'
+    #   I18n.t 'example.one' == 'One interpolation %{foo}'
+    #   I18n.t 'example.two' == 'Two interpolations %{foo} %{bar}'
+    #   I18n.t 'example.one', locale: :other == 'One interpolation %{baz} %{bar}'
+    #
+    # Then we can expect the following results:
+    #   I18n.interpolation_keys('example.zero') #=> []
+    #   I18n.interpolation_keys('example.one') #=> ['foo']
+    #   I18n.interpolation_keys('example.two') #=> ['foo', 'bar']
+    #   I18n.interpolation_keys('one', scope: 'example', locale: :other) #=> ['baz']
+    #   I18n.interpolation_keys('does-not-exist') #=> []
+    def interpolation_keys(key, **options)
+      raise I18n::ArgumentError if !key.is_a?(String) || key.empty?
+
+      return [] unless exists?(key, **options.slice(:locale, :scope))
+
+      translation = translate(key, **options.slice(:locale, :scope))
+
+      raise I18n::NonStringTranslationError unless translation.is_a?(String)
```

Comment:
This _could_ also return an empty `Array`? 🤔 

It's a bit of an odd one: What if you have translations:

```
example_nested:
  one: "%{foo}"
  two: %{bar}"
```

What should `I18n.interpolation_keys('example_nested')` return? `[]`, `[['foo'], ['bar']]`, raise an exception, or ... Something else? 👀